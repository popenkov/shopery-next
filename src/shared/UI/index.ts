export * from './AppLink';
export * from './Buttons';
export * from './Text';
export * from './Logo';
export * from './StarRating';
export * from './Input';
export * from './Radio';
export * from './Checkbox';
export * from './AppSelect';
export * from './Tabs';
export * from './PhoneLink';
export * from './RangeSlider';
export * from './Dropdown';
export * from './Location';
export * from './Accordion';
export * from './BurgerButton';
export * from './CategoriesButton';
