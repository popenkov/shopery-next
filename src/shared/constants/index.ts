export { getRouteHome, getRouteCatalog, PAGE_ROUTES } from './routes';
export { VALIDATION_MESSAGES } from './validation-messages';
