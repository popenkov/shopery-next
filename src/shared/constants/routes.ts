import { AppRouteNames, AppRoutesProps } from 'shared/interfaces/router';

const getRouteHome = () => '/';
const getRouteCatalog = () => '/catalog';
const getRouteProduct = () => '/product/:id';

export const PAGE_ROUTES: AppRoutesProps[] = [
  {
    href: getRouteHome(),
    name: AppRouteNames.HOME,
  },
  {
    href: getRouteCatalog(),
    name: AppRouteNames.CATALOG,
  },
];

export { getRouteHome, getRouteCatalog, getRouteProduct };
