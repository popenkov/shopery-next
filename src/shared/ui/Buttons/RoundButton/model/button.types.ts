export type RoundButtonTheme = 'primary' | 'secondary' | 'ghost';
