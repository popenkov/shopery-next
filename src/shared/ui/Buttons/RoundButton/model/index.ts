export { type TRoundButtonTheme } from './button.types';
