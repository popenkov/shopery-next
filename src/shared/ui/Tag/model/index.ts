export { type TTag, type TTagVariant } from './types';
