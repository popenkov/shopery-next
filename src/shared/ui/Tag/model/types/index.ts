export { type TTag, type TTagVariant } from './tag.types';
