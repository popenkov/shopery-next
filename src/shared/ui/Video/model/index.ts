export { type TVideo } from './video.types';
