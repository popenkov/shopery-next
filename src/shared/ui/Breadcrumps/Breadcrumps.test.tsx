import React from 'react';
import { render } from '@testing-library/react';
import { AppRouteNames } from '@/shared/model';
import { PAGE_ROUTES } from '@/shared/lib/constants/routes';

import { Breadcrumbs } from './';

const crumbs: AppRouteNames[] = [
  AppRouteNames.HOME,
  AppRouteNames.CATALOG,
  AppRouteNames.NOT_FOUND,
];

describe('Breadcrumbs component', () => {
  it('renders correctly', () => {
    const { container } = render(<Breadcrumbs items={crumbs} />);
    expect(container).toMatchSnapshot();
  });

  it('renders home link correctly', () => {
    const { getAllByTestId, debug } = render(<Breadcrumbs items={crumbs} />);
    const homeLink = getAllByTestId('Breadcrumbs.link')[0];
    debug();
    expect(homeLink).toBeInTheDocument();
    expect(homeLink).toHaveAttribute('href', '/'); // assuming getRouteHome() returns '/'
  });

  it('renders breadcrumb items correctly', () => {
    const { getAllByRole, getAllByTestId } = render(<Breadcrumbs items={crumbs} />);
    const breadcrumbItems = getAllByTestId('Breadcrumbs.link');
    expect(breadcrumbItems).toHaveLength(crumbs.length);
    breadcrumbItems.forEach((item, index) => {
      if (index === 0) {
        return;
      }
      expect(item).toHaveTextContent(crumbs[index]);
    });
  });

  it('renders active breadcrumb item correctly', () => {
    const { getAllByTestId } = render(<Breadcrumbs items={crumbs} />);
    const activeItem = getAllByTestId('Breadcrumbs.link')[crumbs.length - 1];
    expect(activeItem).toHaveClass('active');
  });

  it('renders links correctly', () => {
    const { getAllByTestId } = render(<Breadcrumbs items={crumbs} />);
    const links = getAllByTestId('Breadcrumbs.link');

    links.forEach((link, index) => {
      if (index < crumbs.length - 1) {
        const currentPage = PAGE_ROUTES.filter((page) => {
          return page.name === crumbs[index];
        })[0];
        expect(link).toHaveAttribute('href', currentPage.href);
      }
    });
  });
});
