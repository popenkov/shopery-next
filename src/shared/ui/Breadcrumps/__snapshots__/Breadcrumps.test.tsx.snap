// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs component renders correctly 1`] = `
<div>
  <div
    class="breadcrumps"
  >
    <div
      class="breadcrumpsContent"
    >
      <ul
        class="list"
      >
        <li
          class="item"
        >
          <a
            class="link"
            data-testid="breadcrumbs-link"
            href="/"
          >
            <svg
              class="homeIcon"
              data-testid="test-icon"
              fill="none"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M4 10L12 3L20 10V20H15V16C15 15.2044 14.6839 14.4413 14.1213 13.8787C13.5587 13.3161 12.7956 13 12 13C11.2043 13 10.4413 13.3161 9.87868 13.8787C9.31607 14.4413 9 15.2044 9 16V20H4V10Z"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="1.5"
              />
            </svg>
            <svg
              class="chevron"
              data-testid="test-icon"
              fill="none"
              viewBox="0 0 12 12"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M10 4L6 8L2 4"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
              />
            </svg>
          </a>
        </li>
        <li
          class="item"
        >
          <a
            class="link"
            data-testid="breadcrumbs-link"
            href="/catalog"
          >
            Catalog
          </a>
          <svg
            class="chevron"
            data-testid="test-icon"
            fill="none"
            viewBox="0 0 12 12"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M10 4L6 8L2 4"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
            />
          </svg>
        </li>
        <li>
          <a
            class="link active"
            data-testid="breadcrumbs-link"
            href="#"
          >
            404 Error Page
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
